smarts/core/bubble_manager.py

```diff
@@ -271,11 +281,41 @@ class Cursor:
     transition: Optional[BubbleTransition] = None
     bubble: Optional[Bubble] = None
 
+    @classmethod
+    def for_removed(
+        cls,
+        vehicle_id: str,
+        bubble: Bubble,
+        index: VehicleIndex,
+        vehicle_ids_per_bubble: Dict[Bubble, Set[str]],
+    ) -> "Cursor":
+        """Generate a cursor for an inactive bubble.
+        Args:
+            vehicle (Vehicle):
+                The vehicle that is to be tracked.
+            bubble (Bubble):
+                The bubble that the vehicle is interacting with.
+            index (VehicleIndex):
+                The vehicle index the vehicle is in.
+            vehicle_ids_per_bubble (Dict[Bubble, Set[str]]):
+                Bubbles associated with vehicle ids.
+            running_cursors (Set["Cursor"]):
+                A set of existing cursors.
+        """
+        was_in_this_bubble = vehicle_id in vehicle_ids_per_bubble[bubble]
+        is_hijacked, is_shadowed = index.vehicle_is_hijacked_or_shadowed(vehicle_id)
+        transition = None
+        if was_in_this_bubble and (is_shadowed or is_hijacked):
+            transition = BubbleTransition.AirlockExited
```

Comment:
I'm not sure I understand the logic here.  This vehicle has already been removed, right?  Maybe add a comment to explain what's going on...

Reply:
The bubble manager appears to keep track of all vehicles, this is a check that happens just before the vehicle is removed because there is a need to check if the vehicle has left the bubble.